### PR TITLE
chore(deps): upgrade credo, anoncreds, yarn

### PR DESCRIPTION
**Pull request by bryce-mcmath**
Upgraded `yarn` to `4.5.0`, `@credo-ts/*` to `0.5.10`, and `@hyperledger/anoncreds-*` to `0.2.4 `

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -33,4 +33,13 @@ lerna-debug.log*
...
+!.vscode/extensions.json
+
+# yarn
+**/.pnp.*
+**/.yarn/*
+!**/.yarn/patches
+!**/.yarn/plugins
+!**/.yarn/releases
+!**/.yarn/sdks
+!**/.yarn/versions
```
Comment (genaris) [inline .gitignore:45]: These exceptions are not needed anymore, right?

**Review by genaris — APPROVED** [review]
Thanks @bryce-mcmath ! It's great that you found an elegant CI setup for the newer yarn.

File: .gitignore (hunk 1 above)
Comment (bryce-mcmath) [reply .gitignore:45]: You can still add patches and such this way with this version of yarn.

Comment (bryce-mcmath) [reply .gitignore:45]: So they're not needed right now but they might be needed in the future

@genaris let me know if you want me to remove them

Comment (genaris) [reply .gitignore:45]: No problem!

**Pull request merged**